File: customer-service/src/main/java/com/wks/servicemarketplace/customerservice/adapters/db/dao/DefaultEventDao.java
package com.wks.servicemarketplace.customerservice.adapters.db.dao;

import com.wks.servicemarketplace.common.events.EventEnvelope;
import com.wks.servicemarketplace.customerservice.core.daos.EventDao;

import javax.inject.Inject;
import java.sql.Connection;

import static org.jooq.impl.DSL.field;
import static org.jooq.impl.DSL.table;

public class DefaultEventDao extends BaseDAO implements EventDao {

    @Inject
    public DefaultEventDao(DataSource dataSource) {
        super(dataSource);
    }

    @Override
    public boolean saveEvent(Connection connection, EventEnvelope event) {
        return create(connection).insertInto(
                table("events"),
                field("event_uuid"),
                field("event_type"),
                field("event_body"),
                field("entity_id"),
                field("entity_type")
        ).values(
                event.getEventId().toString(),
                event.getEventType().toString(),
                event.getEventBody(),
                event.getEntityId(),
                event.getEntityType()
        ).execute() == 1;
    }
}

File: auth-service/src/main/kotlin/com/wks/servicemarketplace/authservice/adapters/db/dao/DefaultEventDao.kt
package com.wks.servicemarketplace.authservice.adapters.db.dao

import com.wks.servicemarketplace.authservice.core.EventDao
import com.wks.servicemarketplace.common.events.EventEnvelope
import org.jooq.JSON
import org.jooq.impl.DSL.*
import java.sql.Connection
import javax.inject.Inject

class DefaultEventDao @Inject constructor(dataSource: DataSource) : BaseDao(dataSource), EventDao {
    override fun saveEvent(connection: Connection, event: EventEnvelope): Boolean {
        return create(connection).insertInto(table("events"),
                field("event_uuid"),
                field("event_type"),
                field("event_body"),
                field("entity_id"),
                field("entity_type")
        ).values(
                event.eventId.toString(),
                event.eventType.name,
                JSON.valueOf(event.eventBody),
                event.entityId,
                event.entityType
        ).execute() == 1
    }
}